PSModules/AzureApiFunctions.psm1

```diff
@@ -0,0 +1,60 @@
+function ApiRequest {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $Url,
+        [Parameter(Mandatory=$true)]
+        [string] $ApiKey,

```

Comment:
Should this be a securesstring?

Reply:
The api key is read in by Invoke-AzureRmResourceAction (it doesn't get passed into the script). Changing it to securestring would involve encrypting it before passing it to this helper function and then decrypting it in the helper function. I'm not sure this would make the code any more secure.

Reply:
Agreed